faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -399,10 +407,17 @@ public CompletableFuture<HttpResponse<String>> get(String path, Optional<Duratio
    * @param path   the relative path of the resource.
    * @param params a map containing the request parameters.
    * @param queryTimeout the query timeout for the current request.
+   * @param traceId A unique identifier for this query. Adheres to the
+   *                [W3C Trace Context](https://w3c.github.io/trace-context/) spec.
+   * @param tags Key-value pair metadata to associate with this query.
    * @return a {@code CompletableFuture} containing the HTTP response.
    */
-  public CompletableFuture<HttpResponse<String>> get(String path, Map<String, List<String>> params, Optional<Duration> queryTimeout) {
-    return performRequest("GET", path, Optional.empty(), params, queryTimeout);
+  public CompletableFuture<HttpResponse<String>> get(String path,
+                                                     Map<String, List<String>> params,
+                                                     Optional<Duration> queryTimeout,
+                                                     Optional<String> traceId,
+                                                     Optional<Map<String, String>> tags) {
```

Comment:
Perhpas we should overload these methods to make a new version with these new parameters.

That way user's that don't want to use traceId or tags don't have to touch their code when they upgrade the driver.

If we go that route, I wonder if it'd be worth making a request parameters class.

So you'd change these signatures to be something like:

```(java)
public CompletableFuture<HttpRespnose<String>> get(String path, Map<String, List<String>> params, Optional<RequestConfig> requestConfig)
```

that way we don't have to keep adding more parameters and we could embed query timeout, tags, traceparent, and whatever else in there.

Reply:
Fixed